util/etcd/etcd.go

```diff
@@ -0,0 +1,382 @@
+func (c *Client) MoveLeader(transfereeID uint64) error {
+	cli, err := c.newEtcdClient()
+	if err != nil {
+		return err
+	}
+	var lastError error
+	err = wait.ExponentialBackoff(c.backoffParams, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		_, err = cli.MoveLeader(ctx, transfereeID)
+		cancel()
+		if err == nil {
+			return true, nil
+		}
+		c.log.Error(err, "failed to get move etcd member")
+		lastError = err
+		return false, nil
+	})
+	if err != nil {
+		return lastError
+	}
+	return nil
+}
+
+// RemoveMember removes a given member
+func (c *Client) RemoveMember(memberID uint64) error {
+	cli, err := c.newEtcdClient()
+	if err != nil {
+		return err
+	}
+	var lastError error
+	err = wait.ExponentialBackoff(c.backoffParams, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		_, err = cli.MemberRemove(ctx, memberID)
+		cancel()
+		if err == nil {
+			return true, nil
+		}
+		c.log.Error(err, "failed to remove etcd member")
+		lastError = err
+		return false, nil
+	})
+	if err != nil {
+		return lastError
+	}
+	return nil
+}
+
+// UpdateMemberPeerList updates the given member's list of peers
+func (c *Client) UpdateMemberPeerList(id uint64, peerURLs []string) (*[]Member, error) {
+	cli, err := c.newEtcdClient()
+	if err != nil {
+		return nil, err
+	}
+	var lastError error
+	var resp *clientv3.MemberUpdateResponse
+	err = wait.ExponentialBackoff(c.backoffParams, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		resp, err = cli.MemberUpdate(ctx, id, peerURLs)
+		cancel()
+		if err == nil {
+			return true, nil
+		}
+		c.log.Error(err, "failed to get etcd member update list")
+		lastError = err
+		return false, nil
+	})
+	if err != nil {
+		return nil, lastError
+	}
+
+	members := make([]Member, len(resp.Members))
+	for i, m := range resp.Members {
+		members[i] = clientMemberToclientMember(m)
+	}
+
+	return &members, nil
+}
+
+// Retrieves all alarms on a cluster
+func (c *Client) Alarms() (*[]MemberAlarm, error) {
+	cli, err := c.newEtcdClient()
+	if err != nil {
+		return nil, err
+	}
+	var lastError error
+	var resp *clientv3.AlarmResponse
+	err = wait.ExponentialBackoff(c.backoffParams, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		resp, err = cli.AlarmList(ctx)
+		cancel()
+		if err == nil {
+			return true, nil
+		}
+		c.log.Error(err, "failed to get etcd alarm list")
+		lastError = err
+		return false, nil
+	})
+	if err != nil {
+		return nil, lastError
+	}
+
+	memberAlarms := make([]MemberAlarm, len(resp.Alarms))
+	for i, a := range resp.Alarms {
+		memberAlarms[i] = MemberAlarm{
+			MemberID: a.GetMemberID(),
+			Type:     AlarmType(a.GetAlarm()),
+		}
+	}
+
+	return &memberAlarms, nil
+}
+
+// SetBackoffParams replaces the default backoff parameters
+func (c *Client) SetBackoffParams(params wait.Backoff) {
```

Comment:
This comment is out of scope for this PR and could be an improvement later on, but generally I'd suggest using functional options: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis in this kind of scenario